src/lib/storage/fsst_segment.cpp

```diff
@@ -0,0 +1,167 @@
+#include "fsst_segment.hpp"
+
+#include <climits>
+#include <sstream>
+#include <string>
+#include "resolve_type.hpp"
+#include "storage/vector_compression/base_compressed_vector.hpp"
+#include "storage/vector_compression/base_vector_decompressor.hpp"
+#include "storage/vector_compression/resolve_compressed_vector_type.hpp"
+#include "utils/assert.hpp"
+#include "utils/performance_warning.hpp"
+
+#include <libfsst.hpp>
+
+namespace opossum {
+
+template <typename T>
+FSSTSegment<T>::FSSTSegment(pmr_vector<unsigned char>& compressed_values,
+                            std::unique_ptr<const BaseCompressedVector>& compressed_offsets,
+                            pmr_vector<uint64_t>& reference_offsets, std::optional<pmr_vector<bool>>& null_values,
+                            uint64_t number_elements_per_reference_bucket, fsst_decoder_t& decoder)
+    : AbstractEncodedSegment{data_type_from_type<pmr_string>()},
+      _compressed_values{std::move(compressed_values)},
+      _compressed_offsets{std::move(compressed_offsets)},
+      _reference_offsets{std::move(reference_offsets)},
+      _null_values{std::move(null_values)},
+      _number_elements_per_reference_bucket{number_elements_per_reference_bucket},
+      _decoder{decoder},
```

Comment:
Since that's the only thing not moved, does that create a copy?

Reply:
Yes, we believe, it should call the implicitly declared copy constructor of `fsst_decoder_t`. `std::move` would also create a copy, since we get the following warning in clang debug tidy when using `std::move`:

```
/home/Petr.Tsayun/hyrise/src/lib/storage/fsst_segment.cpp:29:16: error: std::move of the variable 'decoder' 
of the trivially-copyable type 'fsst_decoder_t' has no effect; remove std::move() 
[hicpp-move-const-arg,-warnings-as-errors]
      _decoder{std::move(decoder)},
```